### PR TITLE
Fix: Autoplay audio on card previews only when setting is enabled

### LINKED ISSUE #16519: [BUG]: Updated preview system seems to ignore "don't play audio automatically" setting
### Checked for duplicates?

- [X] This issue is not a duplicate

### Does it also happen in the desktop version?

- [X] This bug does not occur in the latest version of Anki

### What are the steps to reproduce this bug?

Create a card with audio, in a deck with the  "don't play audio automatically" setting enabled in the deck options. View the card in the newly updated preview system.

### Expected behaviour

The audio should not play

### Actual behaviour

The audio plays in preview mode, but does not play when actually reviewing the card.

### Debug info

```text
AnkiDroid Version = 2.18.2 (f624ca684f2bee10203c67ed4f801aa4c5423b95)

Backend Version = 0.1.38-anki24.04.1 (24.04.1 ccd9ca1a8309b80bcb50ddc5d99c7ce63440bce9)

Android Version = 13 (SDK 33)

ProductFlavor = play

Manufacturer = Google

Model = Pixel 4a

Hardware = sunfish

Webview User Agent = Mozilla/5.0 (Linux; Android 13; Pixel 4a Build/TQ3A.230805.001.S1; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/124.0.6367.179 Mobile Safari/537.36

ACRA UUID = 2af5d69b-b468-4f7c-844d-e04a95aae12f

FSRS Enabled = false

Crash Reports Enabled = true
```

### (Optional) Anything else you want to share?

I'm actually using the tts audio tag in my version of the cards, plus some overly complicated JavaScript. I'll be testing a more simplified version of the card once I've submitted this ticket and then report back.

I'm also experiencing other problems with the new preview system, but this is the most straightforward, and I'm hoping that once this is resolved my other problems also disappear. Again, I'm using strange JavaScript code and it will be a pain to have to explain it or rework it!

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by leobugeja**
## Purpose / Description

Video demonstration of the card preview only auto playing the audio when the setting is on:
[preview_audio_autoplay_fix.webm](https://github.com/ankidroid/Anki-Android/assets/28263886/c31d166a-06a7-4a93-a4c8-e7197e904d5b)

## Fixes
* Fixes #16519

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/CardMediaPlayer.kt [hunk 1]
```diff
@@ -166,6 +166,14 @@ class CardMediaPlayer : Closeable {
         }
     }
 
+    suspend fun autoplayAllSoundsForSide(cardSide: CardSide): Job? {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/CardMediaPlayer.kt:169]: Why the extra method and not adding the check directly to _playAllSoundsForSide_ ?

**Review by lukstbit — CHANGES_REQUESTED** [review]
Lint is failing. See the details of the Code Quality Checks  below.

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/CardMediaPlayer.kt (hunk 1 above)
Comment (leobugeja) [reply AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/CardMediaPlayer.kt:169]: There are other places which call `playAllSoundsForSide` and should trigger the audio playback even when `config.autoplay` is `false`. For example within `playSounds` in `AbstractFlashcardViewer.kt`, `doAudioReplay` can be `true` while `config.autoplay` is `false` and `playAllSoundsForSide` will still get called (indirectly).

**Review by lukstbit — APPROVED** [review]
LGTM, thank you!

One last nitpick: Please provide a better commit message.

**Review by david-allison — APPROVED** [review]
Ditto: needs a better commit message

**Comment by leobugeja** [comment]
Renamed commit to `fix: card preview autoplay audio`
(my bad I had took the wrong commit message when I squashed)

**Pull request merged**